nixos/lib/test-driver/test_driver/qmp.py

```diff
@@ -0,0 +1,65 @@
+import socket
+from queue import Queue
+from typing import Dict, Any, Iterator
+import json
+from pathlib import Path
+
+
+class QMPSession:
+    def __init__(self, sock: socket.socket) -> None:
+        self.sock = sock
+        self.pending_events: Queue[Dict[str, Any]] = Queue()
+        self.reader = sock.makefile("r")
+        self.writer = sock.makefile("w")
+        hello = self._result()
+        assert "QMP" in hello, f"Unexpected result: {hello}"
+        self.send("qmp_capabilities")
```

Comment:
the docs say that qmp returns an object with the key `error` instead of a `return` key. This remains unchecked here.

Reply:
I made more clear the format for the Greeting message, let me know if I did something wrong.